subsys/bluetooth/controller/ll_sw/ctrl.c

```diff
@@ -1095,7 +1105,71 @@ static u32_t isr_rx_scan_report(u8_t rssi_ready)
 	return 0;
 }
 
-static inline u32_t isr_rx_scan(u8_t irkmatch_id, u8_t rssi_ready)
+static inline bool isr_rx_scan_check(u8_t irkmatch_ok, u8_t devmatch_ok,
+				     u8_t rl_idx)
+{
+#if defined(CONFIG_BLUETOOTH_CONTROLLER_PRIVACY)
+	return ((((_radio.scanner.filter_policy & 0x01) == 0) &&
+		(!devmatch_ok || ctrl_rl_idx_allowed(irkmatch_ok, rl_idx))) ||
```

Comment:
the one I have in the code is correct:

No whitelist:
(!devmatch_ok || ctrl_rl_idx_allowed(irkmatch_ok, rl_idx)))

Either the device is completely unknown (!devmatch) or there was a match but the index is allowed (not in device privacy mode)

Whitelist:
(devmatch_ok) || (ctrl_irk_whitelisted(rl_idx))

Either we hit the whitelist or we had an IRK match that corresponds to a whitelisted device